samza-core/src/main/java/org/apache/samza/config/TaskConfig.java

```diff
@@ -105,6 +105,7 @@
   private static final String BROADCAST_STREAM_PATTERN = "^[\\d]+$";
   private static final String BROADCAST_STREAM_RANGE_PATTERN = "^\\[[\\d]+\\-[\\d]+\\]$";
   public static final String CHECKPOINT_MANAGER_FACTORY = "task.checkpoint.factory";
+  public static final String CHECKPOINT_MANAGER_CONSUMER_STOP_AFTER_FIRST_READ = "task.checkpoint.consumer.stop.after.first.read";
```

Comment:
Why do this as a config instead of a parameter?
If internal config, let's use samza.internal prefix.

Reply:
@prateekm The rationale between @mynameborat and I is that the internal config was a lighter-weight change.

Passing this via construction parameter would mean changing the `CheckpointManagerFactory` API and underlying implementations. If we're ok with that, I'm open to either approach.

Reply:
+1 to internal prefix. We were debating between passing the exposing task mode through parameter vs config. The config route seemed less invasive and also doesn't introduce breaking API changes. 
IIRC, we want to know if the container is standby or not. Given KafkaCheckpointManager is a singleton for the container, is there a way to infer on the fly from the existing configurations if it is standby or active? for e.g. task.inputs is empty?

Reply:
I spoke with @rmatharu and there is nothing in config we can use to infer standby vs not. If we're all ok with the internal config, I will change the prefix and we can go with that.

May also be worth pointing out that this PR doesn't do much to lock us into anything - if we later want to treat this information as more a first class citizen we can go ahead and change the factory API.

Reply:
Is adding a parameter like JobModel or containermodel, etc to CheckpointManagerFactory  a big deal?

Reply:
I agree. For now, the config approach is minimally invasive and backward compatible. We can always reevaluate this if there is a stronger justification to have container model exposed to the checkpoint manager. 
Since it is not a user config, we should be able to deprecate it without much effort.

Reply:
makes sense

Reply:
Closing this thread then. Thanks.